src/OperationResponse.php

```diff
@@ -243,7 +298,7 @@ public function getResult()
      */
     public function getError()
     {
-        if (!$this->isDone() || is_null($this->lastProtoResponse->getError())) {
+        if (!$this->hasProtoResponse() || !$this->isDone()) {
             return null;
         }
         return $this->lastProtoResponse->getError();
```

Comment:
Should this be using the configurable `operationErrorCodeMethod `?

Reply:
Again, I think it would be best to remove that field, but if we keep it, how do you see it being used here?

Reply:
This is updated... Now it returns a `Google\Rpc\Status` object populated with the result of `operationErrorCodeMethod` and `operationErrorMessageMethod`. 

Additionally, the `hasErrors` private method (called in `operationSuccessful` and `operationFailed`) has the following behavior: 
1. checks for the existence of `getError`, if it exists, returns false if it's empty, true otherwise.
2. checks for the existence of `operationErrorCodeMethod`, if it exists, returns false if that returns empty, true otherwise
3. throws an exception saying it cannot determine the error code status (e.g. it's misconfigured)

This means the error annotations are mostly only used to return a consistent object from `OperationResponse::getError`, but _can_ be used to determine the success of the operation _if_ the `error` field doesn't exist.